pkg/storage/etcd/etcd_helper.go

```diff
@@ -56,27 +53,24 @@ func (c *EtcdConfig) GetType() string {
 
 // implements storage.Config
 func (c *EtcdConfig) NewStorage() (storage.Interface, error) {
-	etcdClient := etcd.NewClient(c.ServerList)
-	if etcdClient == nil {
-		return nil, errors.New("Failed to create new etcd client from serverlist")
+	cfg := etcd.Config{
+		Endpoints: c.ServerList,
+		Transport: etcd.DefaultTransport,
+		// TODO: Determine if transport needs optimization
```

Comment:
Transport depended on a forked go-etcd client Dial class, which seemed like a hack.  IMHO we should ping @xiang90 to determine optimized path vs. maintaining a fork of the deprecated interface.